streams/runners/storm/layout/src/main/java/com/hortonworks/streamline/streams/layout/storm/DruidEventMapper.java

```diff
@@ -55,20 +57,27 @@ public DruidEvent(StreamlineEvent event) {
             this.event = event;
         }
 
-        public static boolean isMap(Object map) {
-            return map instanceof Map;
-        }
-
         @Override
         protected Map<String, Object> delegate() {
-            return event;
+            Map<String, Object> updatedEvent = new LinkedHashMap<>();
+            event.forEach((key, value) -> {
+                flattenNestedFields(key, value, updatedEvent, new LinkedList<>(Arrays.asList(key)));
```

Comment:
its better if we skip this method for non nested values to avoid unnecessary overhead. For non-nested, I assume you can just insert the value into the updatedEvent.

If you make the `parentKeyList` a plain String it doesnt have any overhead, so you can leave it as is.